File: Geeker-Admin/src/api/course.ts
import request from '@/utils/request'

export interface CourseVO {
  id?: number
  name: string
  coachId: number
  courtId?: number
  startTime: string
  endTime: string
  maxCount: number
  currentCount?: number
  price: number
  description?: string
  status?: number
  createTime?: string
  updateTime?: string
}

export interface CourseQuery {
  pageNum: number
  pageSize: number
  name?: string
  status?: number
}

export const CourseAPI = {
  getList: (params: CourseQuery) => {
    return request.get('/admin/course/list', params)
  },

  getAll: () => {
    return request.get('/admin/course/all')
  },

  getDetail: (id: number) => {
    return request.get(`/admin/course/${id}`)
  },

  add: (data: CourseVO) => {
    return request.post('/admin/course', data)
  },

  update: (data: CourseVO) => {
    return request.put('/admin/course', data)
  },

  delete: (id: number) => {
    return request.delete(`/admin/course/${id}`)
  },

  enroll: (id: number) => {
    return request.post(`/course/${id}/enroll`)
  },
}

export const CourseClientAPI = {
  getAll: () => {
    return request.get('/courses')
  },

  getById: (id: number) => {
    return request.get(`/courses/${id}`)
  },

  enroll: (id: number) => {
    return request.post(`/courses/${id}/enroll`)
  },
}

File: src/main/java/com/badminton/entity/Course.java
package com.badminton.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("b_course")
public class Course {

    @TableId(type = IdType.AUTO)
    private Integer id;

    private String name;

    @TableField("coach_id")
    private Integer coachId;

    @TableField("court_id")
    private Integer courtId;

    @TableField("start_time")
    private LocalDateTime startTime;

    @TableField("end_time")
    private LocalDateTime endTime;

    @TableField("max_count")
    private Integer maxCount;

    @TableField("current_count")
    private Integer currentCount;

    private BigDecimal price;

    private String description;

    /** 0-下架, 1-开放报名, 2-已满, 3-已结束 */
    private Integer status;

    @TableField(value = "create_time", fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(value = "update_time", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
